server/api/crud/runtimes/nuclio/function.py

```diff
@@ -493,6 +493,20 @@ def _resolve_and_set_build_requirements_and_commands(function, config):
     mlrun.utils.update_in(config, "spec.build.commands", commands)
 
 
+def _enrich_function_node_selector_with_project(
```

Comment:
you are also enriching with service level here
```suggestion
def _enrich_function_node_selector(
```